herbert_core/classes/MPIFramework/@JobDispatcher/private/display_fail_jobs_.m

```diff
@@ -1,107 +1,142 @@
-function    display_fail_jobs_(obj,outputs,n_failed,n_workers,Err_code)
-% Auxiliary method to display job results if the job have
-% failed
+function display_fail_jobs_(obj, outputs, n_failed, n_workers, err_code)
+% Auxiliary method to display job results if the job has failed
+%
 % Input:
-% Outputs -- usually cellarray of the results, returned by a
+% outputs -- usually cellarray of the results, returned by a
 %            parallel job
+%
 % n_failed -- number of tasks failed as the result of parallel
 %             job
+%
 % n_workers-- number of labs used by parallel job initially
 %
-% Err_code -- the text string in the form
+% err_code -- the text string in the form
 %             ERROR_CLASS:error_reason to form identifier of
 %             the exception to throw
-%             If this paraemter is empty, it does not throw anything.
+%             If this parameter is empty, it does not throw anything.
 % Throws:
 % First exception returned from the cluster if such exceptions
-% are present or exception with Err_code as MExeption.identifier
+% are present or exception with err_code as MException.identifier
 % if no errors returned
 %
 
-mEXceptions_outputs = false(size(outputs));
+debug_flag = get(parallel_config, 'debug');
+
+if ~debug_flag
+    log = sprintf("par_fail_%s.log", obj.job_id);
+    fh = fopen(log, 'w');
+end
+
+if ~exist('err_code', 'var') || isempty(err_code)
+    warn_code = 'HORACE:display_fail_jobs:parallel_failure';
+else
+    warn_code = err_code;
+end
+
+MExceptions_outputs = false(size(outputs));
+
 if iscell(outputs)
-    fprintf('Job %s have failed. Outputs: \n',obj.job_id);
+    fprintf('Job %s failed. Outputs: \n', obj.job_id);
+
     for i=1:numel(outputs)
-        if isa(outputs{i},'MException')
-            mEXceptions_outputs(i) = true;
-            fprintf('Task N%d failed. Error %s; Message %s\n',...
-                i,outputs{i}.identifier,outputs{i}.message);
-        elseif isfield(outputs{i},'error') && isa(outputs{i}.error,'MException')
-            mEXceptions_outputs(i) = true;
-            fprintf('Task N%d failed. Reason: %s\n',...
-                i,outputs{i}.fail_reason);
-            
+        if isa(outputs{i}, 'MException')
+            MExceptions_outputs(i) = true;
+            debug_print('Task %d failed. Error %s; Message %s\n', ...
+                    i, outputs{i}.identifier, outputs{i}.message);
+
```

Comment:
I was playing with various failing jobs during the week and would argue that this should be reconsidered. 
When job fails, it prints short list of messages stating worker failed (one row)
Then it prints detailed information. 
The detailed information should go into log, that's correct. 
But short info about job failure should remain. Then it should be one string 
`see file .... for details of the issue`. 

I do not want to investigate this issue too deep, but have a feeling that modified changes are too brief. The suggested solution would provide much better user experience.